### PR TITLE
refactor: upgrade safety

### LINKED ISSUE #531: QA: Ensure all upgradable contracts are upgrade-safe
Things to check:
1. All inherited contracts are initializable and do not rely on constructors
2. All inherited contracts have an appropriately-sized `uint256[X] private __GAP;`
3. All initializers are being called correctly

Output for this Issue:
- A list of all upgradable contracts
- A thorough document with dependency and initialzier relationships

### PR DESCRIPTION
**Pull request by ErinHales**
Closes #531 

Things to check:

 - [x] All inherited contracts are initializable and do not rely on constructors
 - [x] All inherited contracts have an appropriately-sized uint256[X] private __GAP;
 - [x] All initializers are being called correctly

Output for this Issue:

 - [x] A list of all upgradable contracts
 - [x] A thorough document with dependency and initialzer relationships

### CODE REVIEW

File: solidity/optics-core/contracts/governance/GovernanceRouter.sol [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {XAppConnectionManager, TypeCasts} from "../XAppConnectionManager.sol";
 import {IMessageRecipient} from "../../interfaces/IMessageRecipient.sol";
 import {GovernanceMessage} from "./GovernanceMessage.sol";
 
-contract GovernanceRouter is Initializable, IMessageRecipient {
+contract GovernanceRouter is OwnableUpgradeable, IMessageRecipient {
```
Comment (anna-carroll) [inline solidity/optics-core/contracts/governance/GovernanceRouter.sol:14]: do we want `GovernanceRouter` to be `Ownable`? there are no `onlyOwner` functions on the contract, and the true "owner" as it were should be the `governor`

File: solidity/optics-core/contracts/Queue.sol [hunk 2]
```diff
@@ -3,15 +3,18 @@ pragma solidity >=0.6.11;
 
 import "../libs/Queue.sol";
 
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
```
Comment (anna-carroll) [inline solidity/optics-core/contracts/Queue.sol:6]: why is the `QueueManager` `Ownable`? there are no `onlyOwner` functions on the contract

**Review by anna-carroll — COMMENTED** [review]
Some questions about `Ownable`

**Comment by prestwich** [comment]
@ErinHales can you create a chart that shows inheritance for each core contract?

e.g. 
```
--- Home  
  |  
  |- Common
```

File: solidity/optics-core/contracts/governance/GovernanceRouter.sol (hunk 1 above)
Comment (ErinHales) [reply solidity/optics-core/contracts/governance/GovernanceRouter.sol]: Reverted

**Comment by ErinHales** [comment]
> @ErinHales can you create a chart that shows inheritance for each core contract?
> 
> e.g.
> 
> ```
> --- Home  
>   |  
>   |-- Common
>     |
>     |- QueueManager
> ```

located in `contract-inheritance.md`

File: contract-inheritance.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Contract Inheritance
+
+## Core Contracts
+
+- Home
```
Comment (prestwich) [inline contract-inheritance.md:5]: missing `OwnableUpgradable`

File: contract-inheritance.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  - MerkleTreeManager
+  - Common
+    - QueueManager
+- Replica
+  -Common
+    -QueueManager
+- UpdaterManager
```
Comment (prestwich) [inline contract-inheritance.md:12]: missing `Ownable`.

Comment (prestwich) [reply contract-inheritance.md:12]: may be missing all outside dependencies?

Comment (prestwich) [reply contract-inheritance.md:12]: ya, need to include outside dependencies

File: contract-inheritance.md (hunk 3 above)
Comment (ErinHales) [reply contract-inheritance.md]: Oh, thought you only wanted Optics contracts documented.  I'll update

Comment (ErinHales) [reply contract-inheritance.md]: Do you want interfaces documented? Or just contracts?

**Review by prestwich — APPROVED** [review]

File: contract-inheritance.md [hunk 5]
```diff
@@ -9,7 +9,7 @@
   - OwnableUpgradeable
 - Replica (initializes Common)
   -Common (initializes QueueManager)
-    -QueueManager
+  -QueueManager
```
Comment (ErinHales) [inline contract-inheritance.md:12]: Previous is correct

**Pull request merged**